### PR TITLE
shred: add support for octal and hex size

### PR DESCRIPTION
**Pull request by shinhs0506**

### CODE REVIEW

File: src/uucore/src/lib/parser/parse_size.rs [hunk 1]
```diff
@@ -142,10 +164,60 @@ impl<'parser> Parser<'parser> {
             Ok(n) => n,
             Err(_) => return Err(ParseSizeError::size_too_big(size)),
         };
+
+        // parse string into u64
+        let number: u64 = match number_system {
```
Comment (sylvestre) [inline src/uucore/src/lib/parser/parse_size.rs:169]: what about something like:
```

fn parse_number(numeric_string: &str, radix: u32, original_size: &str) -> Result<u64, ParseSizeError> {
    u64::from_str_radix(numeric_string, radix).map_err(|_| ParseSizeError::parse_failure(original_size))
}

let number: u64 = match number_system {
    NumberSystem::Decimal => {
        if numeric_string.is_empty() {
            1
        } else {
            parse_number(&numeric_string, 10, size)?
        }
    }
    NumberSystem::Octal => {
        let trimmed_string = numeric_string.trim_start_matches('0');
        parse_number(trimmed_string, 8, size)?
    }
    NumberSystem::Hexadecimal => {
        let trimmed_string = numeric_string.trim_start_matches("0x");
        parse_number(trimmed_string, 16, size)?
    }
};

```

Comment (shinhs0506) [reply src/uucore/src/lib/parser/parse_size.rs:154]: sounds good!

File: src/uu/shred/src/shred.rs [hunk 2]
```diff
@@ -318,38 +319,17 @@ pub fn uu_app() -> Command {
...
+    match size_str_opt {
+        Some(size) => match parse_size(size.as_str()) {
+            Ok(res) => Some(res),
+            Err(_) => {
+                show_error!("invalid file size: {}", size.quote());
+                std::process::exit(1)
+            }
+        },
+        None => None,
+    }
```
Comment (sylvestre) [inline src/uu/shred/src/shred.rs:332]:
```suggestion
size_str_opt
    .and_then(|size| parse_size(size.as_str()).ok())
    .or_else(|| {
        if let Some(size) = size_str_opt {
            show_error!("invalid file size: {}", size.quote());
            std::process::exit(1);
        }
        None
    })
```

**Comment by sylvestre** [comment]
thanks!

**Comment by sylvestre** [comment]
and bravo for:
`Congrats! The gnu test tests/misc/shred-size is no longer failing!`

**Pull request merged**